src/logic/actions.ts

```diff
@@ -0,0 +1,22 @@
+import { GameState, GameStatus, GameActionType, GameAction } from "./types";
+
+export const initialState: GameState = {
+  status: GameStatus.Initial,
+  board: [],
+};
+
+export function rootReducer<T>(
+  state = initialState,
+  action: GameAction<T>
+): GameState {
+  switch (action.type) {
+    case GameActionType.StartGame:
+      return { ...state, status: GameStatus.Running };
+    default:
+      return state;
+  }
+}
+
+export function startGame(): GameAction<void> {
```

Comment:
I also usually throw the reducer into its own file (logic/reducer.ts for example) and keep the actions in their own file. As both can get large.